framework/Optimizers/RavenSampled.py

```diff
@@ -0,0 +1,566 @@
+    self.raiseAMessage('*'*80)
+    self.raiseAMessage('Optimizer Final Results:')
+    self.raiseADebug('')
+    self.raiseADebug(' - Trajectory Results:')
+    self.raiseADebug('  TRAJ   STATUS    VALUE')
+    statusTemplate = '   {traj:2d}  {status:^11s}  {val: 1.3e}'
+    # print cancelled traj
+    for traj, info in self._cancelledTraj.items():
+      val = info['value']
+      status = info['reason']
+      self.raiseADebug(statusTemplate.format(status=status, traj=traj, val=s * val))
+    # check converged traj
+    for traj, info in self._convergedTraj.items():
+      opt = self._optPointHistory[traj][-1][0]
+      val = info['value']
+      self.raiseADebug(statusTemplate.format(status='converged', traj=traj, val=s * val))
+      if bestValue is None or val < bestValue:
+        bestTraj = traj
+        bestValue = val
+    # further check active unfinished trajectories
+    for traj in self._activeTraj:
+      opt = self._optPointHistory[traj][-1][0]
+      val = opt[self._objectiveVar]
+      self.raiseADebug(statusTemplate.format(status='active', traj=traj, val=s * val))
+      if bestValue is None or val < bestValue:
+        bestValue = val
+        bestTraj = traj
+    bestOpt = self.denormalizeData(self._optPointHistory[bestTraj][-1][0])
+    bestPoint = dict((var, bestOpt[var]) for var in self.toBeSampled)
+    self.raiseADebug('')
+    self.raiseAMessage(' - Final Optimal Point:')
+    finalTemplate = '    {name:^20s}  {value: 1.3e}'
+    finalTemplateInt = '    {name:^20s}  {value: 3d}'
+    self.raiseAMessage(finalTemplate.format(name=self._objectiveVar, value=s * bestValue))
+    self.raiseAMessage(finalTemplateInt.format(name='trajID', value=bestTraj))
+    for var, val in bestPoint.items():
+      self.raiseAMessage(finalTemplate.format(name=var, value=val))
+    self.raiseAMessage('*'*80)
+    # write final best solution to soln export
+    self._updateSolutionExport(bestTraj, self.normalizeData(bestOpt), 'final')
+
+  ###################
+  # Utility Methods #
+  ###################
+  def incrementIteration(self, traj):
+    """
+      Increments the "generation" or "iteration" of an optimization algorithm.
+      The definition of generation is algorithm-specific; this is a utility for tracking only.
+      @ In, traj, int, identifer for trajectory
+      @ Out, None
+    """
+    self.__stepCounter[traj] += 1
+
+  def getIteration(self, traj):
+    """
+      Provides the "generation" or "iteration" of an optimization algorithm.
+      The definition of generation is algorithm-specific; this is a utility for tracking only.
+      @ In, traj, int, identifer for trajectory
+      @ Out, counter, int, iteration of the trajectory
+    """
+    return self.__stepCounter[traj]
+
+  # * * * * * * * * * * * *
+  # Constraint Handling
+  def _handleExplicitConstraints(self, proposed, previous, pointType):
+    """
+      Considers all explicit (i.e. input-based) constraints
+      @ In, traj, int, identifier
```

Comment:
docstring is not accurate (no traj).proposed is mentioned twice

Reply:
fixed